File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/methodadapter/tree/TreeOverwritePatternMatcher.java
package net.yakclient.mixin.base.internal.methodadapter.tree;

import net.yakclient.mixin.base.internal.instruction.Instruction;
import org.objectweb.asm.tree.InsnList;

/**
 * Defines the pattern for overwriting the method and inserting the mixin
 * as the new method.
 * <p>
 *
 * Note: This can cause serious issues with multiple mixins clashing. Use this
 * sparingly.
 *
 * <p>
 * Type: Overwrite
 *
 * @author Durgan McBroom
 */
public class TreeOverwritePatternMatcher extends TreeMixinPatternMatcher {
    public TreeOverwritePatternMatcher(Instruction instructions) {
        super(instructions);
    }

    @Override
    public void transform(InsnList insn) {
        insn.clear();
        this.insertInsn(insn);
    }
}

File: base/src/main/kotlin/net/yakclient/mixins/base/internal/methodadapter/tree/TreeOverwritePatternMatcher.kt
package net.yakclient.mixins.base.internal.methodadapter.tree

import net.yakclient.mixins.base.internal.instruction.Instruction
import org.objectweb.asm.tree.InsnList

/**
 * Defines the pattern for overwriting the method and inserting the mixin
 * as the new method.
 *
 *
 *
 * Note: This can cause serious issues with multiple mixins clashing. Use this
 * sparingly.
 *
 *
 *
 * Type: Overwrite
 *
 * @author Durgan McBroom
 */
class TreeOverwritePatternMatcher(instructions: Instruction?) : TreeMixinPatternMatcher(instructions) {
    override fun transform(insn: InsnList) {
        insn.clear()
        this.insertInsn(insn)
    }
}